### PR TITLE
Update repository layout with tests for shared libraries

### PR DESCRIPTION
**Pull request by vojtechsokol**

### CODE REVIEW

File: docs/source/repository-dir-layout.md [hunk 1]
```diff
@@ -23,8 +25,16 @@
 
  libraries/                         # Libraries that are shared with all actors.
     common.py                       # These can be modules
-    sharedpkg/                      # or packages.
+    tests/                          # with tests stored here.
+        files/                      # If tests need to use some mocked files, they should be placed here
+                                    # and refenreced from the tests using 'relative path, i.e. 'tests/files'
```
Comment (jmikovic) [inline docs/source/repository-dir-layout.md]:
```suggestion
                                    # and refenreced from the tests using 'relative path', i.e. 'tests/files'
```

File: docs/source/repository-dir-layout.md [hunk 2]
```diff
@@ -23,8 +25,16 @@
...
+        test_common.py
+    sharedpkg/                      # Or they can be packages
         __init__.py
+        tests/                      # with tests stored here.
+            test_sharedpkg.py
+            files/                  # If tests need to use some mocked files, they should be placed here
+                                    # and refenreced from the tests using 'absolute path' 'sharedpkg/tests/files'
```
Comment (jmikovic) [inline docs/source/repository-dir-layout.md]:
```suggestion
                                    # and refenreced from the tests using 'relative path' 'sharedpkg/tests/files'
```

Comment (vojtechsokol) [reply docs/source/repository-dir-layout.md]: This should really be 'absolute' path (note the quotation marks). Absolute in the meaning that `sharedpkg` needs to be included in the path, as opposed to 'reative' path (used e.g. in actors), where `tests/files` is used.

Comment (jmikovic) [reply docs/source/repository-dir-layout.md]: well, `sharepkg` doesn't really make it absolute...

but I will let others decide

Comment (Rezney) [reply docs/source/repository-dir-layout.md]: and what about just using word "path"? Just suggestion to move it forward :)

E.g.:
`and refenreced from the tests using 'sharedpkg/tests/files' path?`

Just a small nit: there seems to be a typo in `refenreced`

File: docs/source/unit-testing.md [hunk 3]
```diff
@@ -200,3 +202,33 @@ Note: Dependencies defined the way mentioned above is for test execution only.
...
+
+To run tests of shared libraries (i.e. libraries stored in
+`repositories/system_upgrade/el7toel8/libraries`) environment variable
+`LEAPP_TESTED_LIBRARY` with path of the shared library needs to be specified for
+_pytest_ to be able to import leapp modules.
+
+
+To run tests for all shared libraries:
+```sh
+LEAPP_TESTED_LIBRARY="$PWD/libraries" pytest -sv libraries/tests/test_tcpwrappersutils.py
```
Comment (jmikovic) [inline docs/source/unit-testing.md]: what is the difference between this call and the one bellow? shouldn't the path be `libraries/tests` only?

File: docs/source/repository-dir-layout.md (hunk 2 above)
Comment (vojtechsokol) [reply docs/source/repository-dir-layout.md]: Fixed

File: docs/source/unit-testing.md (hunk 3 above)
Comment (vojtechsokol) [reply docs/source/unit-testing.md]: Good catch, fixed.

**Pull request merged**